packages/webpack-cli/lib/bootstrap.js

```diff
@@ -38,74 +30,72 @@ async function runCLI(cli, commandIsUsed) {
-                    const oldMapValue = argsMap[arg];
-                    argsMap[arg] = {
-                        value: process.argv[idx],
-                        pos: idx,
-                    };
-                    // Swap idx of overridden value
-                    if (oldMapValue) {
-                        argsMap[arg].pos = oldMapValue.pos;
-                        keysToDelete.push(idx + 1);
+                    if (!unknown.startsWith('-')) {
+                        entry.push(unknown);
                     }
                 });
-                // Filter out the value for the overridden key
-                const newArgKeys = Object.keys(argsMap).filter((arg) => !keysToDelete.includes(argsMap[arg].pos));
-                // eslint-disable-next-line require-atomic-updates
-                process.argv = newArgKeys;
-                args = argParser('', core, process.argv);
-                await cli.run(args.opts, core);
-                process.stdout.write('\n');
-                logger.warn('Duplicate flags found, defaulting to last set value');
-            } else {
-                logger.error(err);
-                return;
             }
+        } else if (parsedArgs.unknownArgs.length > 0) {
+            parsedArgs.unknownArgs.forEach((unknown) => {
+                logger.warn('Unknown argument:', unknown);
+            });
+            cliExecuter();
+            return;
+        }
+        const parsedArgsOpts = parsedArgs.opts;
+        if (entry) {
+            parsedArgsOpts.entry = entry;
+        }
+        const result = await cli.run(parsedArgsOpts, core);
+        if (!result) {
+            return;
+        }
+    } catch (err) {
+        if (err.name === 'UNKNOWN_VALUE') {
+            logger.error(`Parse Error (unknown argument): ${err.value}`);
+            return;
+        } else if (err.name === 'ALREADY_SET') {
+            const argsMap = {};
+            const keysToDelete = [];
+            process.argv.forEach((arg, idx) => {
+                const oldMapValue = argsMap[arg];
+                argsMap[arg] = {
+                    value: process.argv[idx],
+                    pos: idx,
+                };
+                // Swap idx of overridden value
+                if (oldMapValue) {
+                    argsMap[arg].pos = oldMapValue.pos;
+                    keysToDelete.push(idx + 1);
+                }
+            });
+            // Filter out the value for the overridden key
+            const newArgKeys = Object.keys(argsMap).filter((arg) => !keysToDelete.includes(argsMap[arg].pos));
+            // eslint-disable-next-line require-atomic-updates
+            process.argv = newArgKeys;
+            args = argParser('', core, process.argv);
+            await cli.run(args.opts, core);
+            process.stdout.write('\n');
+            logger.warn('Duplicate flags found, defaulting to last set value');
+        } else {
+            logger.error(err);
+            return;
```

Comment:
I am afraid for this place, we don't set `exitCode` here, so errors will have `exitCode: 0`, also `eslint-loader`/etc will be broken, can you investigate it and add couple tests in next PRs?

Reply:
Sure